landoapi/api/revisions.py

```diff
@@ -53,6 +62,29 @@ def _format_revision(
     Returns:
         A hash of the formatted revision information.
     """
+    # Load the revision's diff if requested.
```

Comment:
This function is getting pretty large, and collecting distinct blocks of functionality that have a single comment describing them - those are signs it might be time to break this function up into a series of well-named functions.

(something I quickly scratched out, could probably be made even cleaner)
```python
if include_diff:
    response.update(build_response_diff(...))
if include_parents:
    response.update(build_response_parents(...))
if last_author:
    response.update(build_response_last_author(...))
...
```